static/app/views/alerts/rules/metric/insightsMetricField.tsx

```diff
@@ -0,0 +1,221 @@
+import {Fragment, useCallback, useEffect, useMemo} from 'react';
+import styled from '@emotion/styled';
+
+import Tag from 'sentry/components/badge/tag';
+import SelectControl from 'sentry/components/forms/controls/selectControl';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {MetricMeta, ParsedMRI} from 'sentry/types/metrics';
+import type {Project} from 'sentry/types/project';
+import {getDefaultAggregation} from 'sentry/utils/metrics';
+import {getReadableMetricType} from 'sentry/utils/metrics/formatters';
+import {
+  DEFAULT_INSIGHTS_METRICS_ALERT_FIELD,
+  DEFAULT_INSIGHTS_MRI,
+  formatMRI,
+  MRIToField,
+  parseField,
+  parseMRI,
+} from 'sentry/utils/metrics/mri';
+import {useVirtualizedMetricsMeta} from 'sentry/utils/metrics/useMetricsMeta';
+import {middleEllipsis} from 'sentry/utils/string/middleEllipsis';
+import {
+  INSIGHTS_METRICS,
+  INSIGHTS_METRICS_OPERATIONS,
+} from 'sentry/views/alerts/rules/metric/utils/isInsightsMetricAlert';
+
+interface Props {
+  aggregate: string;
+  onChange: (value: string, meta: Record<string, any>) => void;
+  project: Project;
+}
+
+// We actually only store a few aggregations for Insights metrics.
+// The `metrics/meta/` endpoint doesn't know this, so hardcode supported aggregations for now.
+const OPERATIONS = [
+  {
+    label: 'avg',
+    value: 'avg',
+  },
+  {
+    label: 'sum',
+    value: 'sum',
+  },
+  {
+    label: 'min',
+    value: 'min',
+  },
+  {
+    label: 'max',
+    value: 'max',
+  },
+  ...INSIGHTS_METRICS_OPERATIONS,
+];
+
+function aggregateRequiresArgs(aggregation?: string) {
+  return aggregation !== 'spm';
+}
+
+function InsightsMetricField({aggregate, project, onChange}: Props) {
+  const {data: meta, isLoading} = useVirtualizedMetricsMeta(
+    {projects: [parseInt(project.id, 10)]},
+    ['spans']
+  );
+
+  const metaArr = useMemo(() => {
+    return meta
+      .map(
+        metric =>
+          ({
+            ...metric,
+            ...parseMRI(metric.mri),
+          }) as ParsedMRI & MetricMeta
+      )
+      .filter(metric => INSIGHTS_METRICS.includes(metric.mri));
```

Comment:
`INSIGHTS_METRICS` includes `d:spans/exclusive_time@millisecond` - should we drop it from the list - if i recall correctly, we've changed everything over to duration? especially because most of the insights metrics are leaf spans anyway

Reply:
Hmm I still see `span.self_time`/`d:spans/exclusive_time@millisecond` being used in some insights module code, so I think we'll have to keep it here for now. I think we can let this sit for a bit and drop this mri once we have time to confirm we're completely no longer using it